src/App.js

```diff
@@ -1,19 +1,54 @@
 import React from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog'
 import chatMessages from './data/messages.json';
+import {useState} from 'react';
+// import ChatEntry from './components/ChatEntry'
 
-const App = () => {
-  return (
-    <div id="App">
+
+console.log(chatMessages);
+
+function App () {
+
+  const [chatData, setChatData] = useState(chatMessages)
+  const updateChatData = updatedChat => {
+    console.log({updatedChat})
+    const chats = chatData.map(chat => {
+      if(chat.id === updatedChat.id){
+        
+        return updatedChat;
+      } else{
+        return chat;
+      }
+      });
+    setChatData(chats);
+  };
+ 
+  const calcTotalHearts = ()=>{
```

Comment:
great way to use existing data to calculate total likes. You could also use reduce like
```javscript 
    return messageData.reduce((totalLikes, message) => {
      // If messages.liked is true add 1 to totalLikes, else add 0
      return (totalLikes += message.liked ? 1 : 0);
    }, 0); // The 0 here sets the initial value of totalLikes to 0
```